### PR TITLE
feature/marked entity update delete

### PR DESCRIPTION
**Pull request by jamestjw**
- Add update and delete functionality to marked entities
- Prevent updating of marked entities if instructor does not teach a course or if the marked entity is past its submission date
- Add delete functionality of instructor marked entity files

- Allow instructors to add new files after creation of marked entities

### CODE REVIEW

File: public/marked_entities/update.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    die("Invalid MarkedEntity");
+}
+
+// Check if the user is indeed an instructor of this course
+if (is_null(CourseOfferingInstructor::find_by(["offering_id" => $marked_entity->course_offering_id, "user_id"=> $_SESSION["current_user_id"]]))) {
+    die("Course instructor does not teach this course.");
+}
+
+if ($marked_entity->due_date_passed()) {
+    die("Unable to update marked entity pass its due date.");
```
Comment (chris7neves) [inline public/marked_entities/update.php:29]: should be past* its due date 😛

File: public/marked_entities/delete.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?php require_once(dirname(__FILE__)."/../../modules/ensure_logged_in.php"); ?>
+
+<?php
+
+require_once "../../modules/models/marked_entity.php";
+require_once "../../common.php";
+
+$UPDATABLE_FIELDS = ["title", "description", "due_at"];
```
Comment (chris7neves) [inline public/marked_entities/delete.php:8]: should an instructor also be able to attach a file to the marked entity? like a pdf or code file?

**Review by chris7neves — CHANGES_REQUESTED** [review]

File: public/announcements.php [hunk 3]
```diff
@@ -20,8 +19,7 @@
     } catch (PDOException $error) {
         echo "<br>" . $error->getMessage();
     }
-}
-elseif (get_current_role() != "admin"){
+} elseif (get_current_role() != "admin") {
```
[the remark this answers is not in the document]
Comment (jamestjw) [reply public/announcements.php:22]: @chris7neves If I recall correctly, the PDF mentioned that these system wide announcements can only be sent by Admins. Course level announcements aren't a thing

File: public/marked_entities/update.php (hunk 1 above)
Comment (jamestjw) [reply public/marked_entities/update.php:29]: ah yes that's right 😂

File: public/account_settings.php [hunk 4]
```diff
@@ -63,33 +63,32 @@
...
+        empty($_POST["email"]) ?: $user->email = $_POST["email"];
 
-  empty($_POST["first_name"]) ?: $user->first_name = $_POST["first_name"];
-  empty($_POST["last_name"]) ?: $user->last_name = $_POST["last_name"];
-  empty($_POST["email"]) ?: $user->email = $_POST["email"];
+        if (!empty($_POST["student_id"])) {
+            if (ctype_digit($_POST["student_id"])) {
+                $user->student_id = $_POST["student_id"];
+            } else {
+                echo "The entered student ID - '".escape($_POST["student_id"]."' is not valid and should only contain digits.");
```
Comment (chris7neves) [inline public/account_settings.php:74]: why are my account_settings changes here?

File: install.php [hunk 5]
```diff
@@ -16,7 +16,13 @@
...
     $connection->exec($sql);
 
-    echo "Database and table users created successfully.";
+    echo "Database and table created successfully.\n";
+
+    $sql = file_get_contents("data/seed.sql");
+    $sql = sprintf($sql, $dbname);
+    $connection->exec($sql);
+
+    echo "Database seeded successfully.\n";
```
[the remark this answers is not in the document]
Comment (jamestjw) [reply install.php:25]: well we are supposed to populate the database with some initial data for the demo right, so I thought that it'd be useful to get started on that (okay actually the main reason I did it was to make testing easier)

File: public/account_settings.php (hunk 4 above)
Comment (jamestjw) [reply public/account_settings.php:74]: I ran the linter, I guess it fixed a bunch of stuff in your code

**Review by chris7neves — APPROVED** [review]

**Pull request merged**